x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java

```diff
@@ -332,15 +335,21 @@ public StreamInput streamInput() throws IOException {
                         blobContainer.writeBlob(request.blobName, bytesReference, failIfExists);
                     }
                 } else {
-                    blobContainer.writeBlob(
-                        request.blobName,
-                        repository.maybeRateLimitSnapshots(
-                            new RandomBlobContentStream(content, request.getTargetLength()),
-                            throttledNanos::addAndGet
-                        ),
-                        request.targetLength,
-                        failIfExists
-                    );
+                    cancellableThreads.execute(() -> {
+                        try {
+                            blobContainer.writeBlob(
+                                request.blobName,
+                                repository.maybeRateLimitSnapshots(
+                                    new RandomBlobContentStream(content, request.getTargetLength()),
+                                    throttledNanos::addAndGet
+                                ),
+                                request.targetLength,
+                                failIfExists
+                            );
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
```

Comment:
An alternative here instead of re-throwing with a RuntimeException would be to make the org.elasticsearch.common.util.CancellableThreads.Interruptible interface to throw also an IOException or an Exception in general instead of only an InterruptedException (that it currently allows). Feedback is welcome on this point.